### PR TITLE
Fix exotics in loadouts, plus The Life Exotic handling

### LINKED ISSUE #749: Having an Exotic in an Armor Slot Breaks Loadout with Exotic
I have a loadout with an exotic helmet marked as equipped.  If I have an exotic chest piece currently on the character, the loadout will try to 

1. Move all the items to the character.
2. Call equip all to put the gear marked 'equipped' on the character.

Since I have an exotic item on, it fails.

Subsequently applying the Loadout to the character, it will remove the exotic chest piece and then the equip all will succeed.

I should note that the I do have a chest piece in the loadout, but it is not marked as equipped.

### PR DESCRIPTION
**Pull request by bhollis**
Fixed an issue where you could not equip a loadout that included an exotic item when you already had an exotic equipped that was not going to be replaced by the loadout. Also, in general better handling of items with "The Life Exotic" perk.

Fixes #749.

### CODE REVIEW

File: app/scripts/services/dimStoreService.factory.js [hunk 1]
```diff
@@ -120,6 +120,10 @@
       },
       canBeInLoadout: function() {
         return this.equipment || this.type === 'Material' || this.type === 'Consumable';
+      },
+      // "The Life Exotic" Perk on Exotic Items means you can equip another exotic
+      hasLifeExotic: function() {
+        return this.isExotic && this.talentGrid && (_.find(this.talentGrid.nodes, { name: 'The Life Exotic' }) !== undefined);
```
Comment (SunburnedGoose) [inline app/scripts/services/dimStoreService.factory.js]: Will internationalization affect this?

File: app/scripts/services/dimItemService.factory.js [hunk 2]
```diff
@@ -251,18 +254,12 @@
...
-        return _.find(item.talentGrid.nodes, { name: 'The Life Exotic' }) !== undefined;
-      }
-
       if (equippedExotics.length === 0) {
         deferred.resolve(true);
       } else if (equippedExotics.length === 1) {
         var equippedExotic = equippedExotics[0];
 
-        if (hasLifeExotic(item) || hasLifeExotic(equippedExotic)) {
+        if (item.hasLifeExotic() || item.hasLifeExotic()) {
```
Comment (SunburnedGoose) [inline app/scripts/services/dimItemService.factory.js]: Is this supposed to check the same item twice?

File: app/scripts/services/dimStoreService.factory.js (hunk 1 above)
Comment (bhollis) [inline app/scripts/services/dimStoreService.factory.js]: Yes

File: app/scripts/services/dimItemService.factory.js (hunk 2 above)
Comment (bhollis) [inline app/scripts/services/dimItemService.factory.js]: Good catch!

**Pull request merged**